apps/frontend/src/hooks/aave/useAaveReservesHistory.tsx

```diff
@@ -0,0 +1,152 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import axios from 'axios';
+import dayjs from 'dayjs';
+
+import { AAVE_CONTRACT_ADDRESSES } from '../../constants/aave';
+
+export enum ESupportedTimeRanges {
+  OneMonth = '1m',
+  ThreeMonths = '3m',
+  SixMonths = '6m',
+  OneYear = '1y',
+  TwoYears = '2y',
+  FiveYears = '5y',
+}
+
+export const reserveRateTimeRangeOptions = [
+  ESupportedTimeRanges.OneMonth,
+  ESupportedTimeRanges.SixMonths,
+  ESupportedTimeRanges.OneYear,
+];
+export type ReserveRateTimeRange = typeof reserveRateTimeRangeOptions[number];
+
+type RatesHistoryParams = {
+  from: number;
+  resolutionInHours: number;
+};
+
+type APIResponse = {
+  liquidityRate_avg: number;
+  variableBorrowRate_avg: number;
+  stableBorrowRate_avg: number;
+  utilizationRate_avg: number;
+  x: { year: number; month: number; date: number; hours: number };
+};
+
+const requestCache = new Map<string, Promise<APIResponse[]>>();
+const fetchStats = async (
+  reserveId: string,
+  timeRange: ReserveRateTimeRange,
+  endpointURL: string,
+): Promise<APIResponse[]> => {
+  const { from, resolutionInHours } = resolutionForTimeRange(timeRange);
+  const qs = `reserveId=${reserveId}&from=${from}&resolutionInHours=${resolutionInHours}`;
+  const url = `${endpointURL}?${qs}`;
+
+  if (requestCache.has(url)) {
+    return requestCache.get(url)!;
+  }
+
+  const requestPromise = axios
+    .get<APIResponse[]>(url)
+    .then(response => response.data)
+    .finally(() => {
+      requestCache.delete(url);
+    });
+
+  requestCache.set(url, requestPromise);
+
+  return requestPromise;
+};
+
+const resolutionForTimeRange = (
+  timeRange: ReserveRateTimeRange,
+): RatesHistoryParams => {
+  switch (timeRange) {
+    case ESupportedTimeRanges.OneMonth:
+      return {
+        from: dayjs().subtract(30, 'day').unix(),
```

Comment:
maybe makes sense to update `dayjs().subtract(30, 'day')` to `dayjs().subtract(1, 'month')` to handle full months for `OneMonth`